### ISSUE TITLE
Open3D development wheels are unavailable

### ISSUE THREAD

**Issue by AlexanderRitter02** [isl-org/Open3D#6870]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](https://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](https://www.open3d.org/docs/release/) and the [latest documentation](https://www.open3d.org/docs/latest/) (for `main` branch).

### Steps to reproduce the issue

As suggested in the documentation at https://www.open3d.org/docs/release/getting_started.html#development-version-pip, I tried installing the latest development release with:
`pip install -U -f https://www.open3d.org/docs/latest/getting_started.html open3d`

However, this results in the following 404 error:
```
Looking in links: https://www.open3d.org/docs/latest/getting_started.html
Collecting open3d
  ERROR: HTTP error 404 while getting https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip (from https://www.open3d.org/docs/latest/getting_started.html)
ERROR: Could not install requirement open3d from https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip because of HTTP error 404 Client Error: Not Found for url: https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip for URL https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip (from https://www.open3d.org/docs/latest/getting_started.html)
```

Apparently in March there has been a similar issue which was fixed in PR #6694, but this re-occured now.

### Error message

```shell
Looking in links: https://www.open3d.org/docs/latest/getting_started.html
Collecting open3d
  ERROR: HTTP error 404 while getting https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip (from https://www.open3d.org/docs/latest/getting_started.html)
ERROR: Could not install requirement open3d from https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip because of HTTP error 404 Client Error: Not Found for url: https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip for URL https://github.com/isl-org/Open3D/releases/download/devel-main/open3d-0.18.0%2Bafb23f8-app-windows-amd64.zip (from https://www.open3d.org/docs/latest/getting_started.html)
```

### Open3D, Python and System information

```markdown
- Operating system: Windows 10 64-bit
- Python version: Python 3.11.9
- Open3D version: afb23f8
- System architecture: x86
- Is this a remote workstation?: no
- ~How did you install Open3D?: pip~
```

**Comment by ssheorey** [comment]
Thanks for reporting. There's a typo in the URLs, Will fix shortly...

devel-main -> main-devel

Meanwhile, the wheels are available from github releases.

**Comment by timohl** [comment]
I think this got fixed in #6883 and #6890.
